pkg/auth/authorizer/abac/abac.go

```diff
@@ -52,7 +53,46 @@ func (p policyLoadError) Error() string {
 type policyList []*api.Policy
 
 // TODO: Have policies be created via an API call and stored in REST storage.
-func NewFromFile(path string) (policyList, error) {
+func NewFromFile(path string) (*policyList, error) {
+
+	pl, err := newFromFile(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create an empty watcher
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	// Watch the policy file
+	err = watcher.Add(path)
+	if err != nil {
+		return nil, err
+	}
+
+	// On write operations on the policy file, reload policy file
+	go func() {
+		for {
+			select {
+			case ev := <-watcher.Events:
+				switch ev.Op {
+				case fsnotify.Write:
```

Comment:
Yeah, I agree, fsnotify just over complicates things here. 

> You still need to handle the case of an incorrect file. Should you crash, as is currently the case when an incorrect file is provided? Or keep using the old one? How to surface this error to the user?

I don't think we should crash when there's an error due to policy reloading as it could put a running cluster to a halt. Keep using the old one but throw out an error (glog.Errorf(...)) is probably a nicer way of keeping the cluster running while informing the admin who's making the change that there's an error to be corrected. On a successful load, we should also do a glog.Infof(...) so the admin can verify that the change has taken effect. The verification step of checking the log should be documented. Is this reasonable?